### PR TITLE
mostly travis

### PR DESCRIPTION
**Pull request by terminaldweller**
the travis build.
a script to run the tests locally whithout manual intervention.
added some changes to index.js so things work on travis and by the shell script.
added a small section in the readme for running the tests.
all that's left is for someone to add the repo badge(since i havent added the repo to travis i dunno where it is).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -22,9 +22,30 @@ install:
   - source ~/.bashrc
   - git clone https://github.com/ethereum/go-ethereum
   - sed -i 's/ctx, cancel = context\.WithTimeout(ctx, time\.Second\*5)/ctx, cancel = context\.WithTimeout(ctx, time\.Second\*500)/' ./go-ethereum/internal/ethapi/api.go
-    
-  - travis_wait 40 make -C go-ethereum geth
+  - travis_wait 40 make -C go-ethereum
   - sudo cp ./go-ethereum/build/bin/geth /usr/local/bin/geth
+  - stty tostop
+  - sudo iptables -L
+  - sudo iptables -A INPUT -p tcp --dport 8545 -j ACCEPT
```
Comment (chriseth) [inline .travis.yml:29]: OMG this is really required here?

File: .travis.yml [hunk 2]
```diff
@@ -22,9 +22,30 @@ install:
...
+  - sudo ufw allow 8545
+  - sudo iptables -L
+  - echo ilovescrypt > ./password.txt
+  - geth --datadir /tmp/ethereum_dev_mode --password /home/travis/build/bloodstalker/scrypt-interactive/password.txt account new
```
Comment (chriseth) [inline .travis.yml]: I don't think the path here is portable.

File: README.md [hunk 3]
```diff
@@ -1,10 +1,20 @@
...
+For more context on how relays work, look at the [BTCRelay](https://github.com/ethereum/btcrelay) project.<br/>
 
-It enables a step-by-step running and verification of the [Scrypt](https://en.wikipedia.org/wiki/Scrypt) hashing function on the EVM.
+## Running the Tests
 
-This can form the basis of a Dogecoin/Ethereum relay. Dogecoin uses Scrypt for its Proof-of-Work algorithm, specifically to calculate the block hash. Therefore, an ability to compute Scrypt hashes is required in order to verify the validity of Dogecoin transactions, as it allows one to check Merkle proofs from a known block header. This computation is too large given Ethereum's gas constraints, making it an ideal use case for Truebit.
+To run the tests you need `geth` and a JS `node` version higher than 7.6. After getting those you can just run `run.sh` in the roor directory of the repo.<br/>
```
Comment (chriseth) [inline README.md]: Typo: `roor`

File: .travis.yml [hunk 4]
```diff
@@ -22,9 +22,30 @@ install:
...
+  - cat ./password.txt
+  - geth account list
+  - ls
+  - pwd
+  - primaccount=$(geth account list | gawk '{print $3}' | sed 's/{/"/'|sed 's/}/"/')
+  - echo $primaccount
+  - geth --verbosity 6 --dev --rpc --rpcapi admin,miner,eth,net,web3,personal --rpcaddr "localhost" --rpcport "8545" --port "30303" --datadir /tmp/ethereum_dev_mode &>/dev/null &
+  - geth --exec 'loadScript("./test/gethconfig.js")' attach http://127.0.0.1:8545
```
Comment (chriseth) [inline .travis.yml:41]: attaching via IPC does not work?

File: .travis.yml (hunk 1 above)
Comment (terminaldweller) [reply .travis.yml:29]: no but it was a really painful geth setup. the ubuntu image travis gives out is quite minimalistic. at a certain point i suspected there was something wrong with the iptables. i left some debug stuff inside the travis build script so maybe in the future when things change, they wont be as painful.

File: README.md (hunk 3 above)
Comment (terminaldweller) [reply README.md]: kk

File: .travis.yml (hunk 4 above)
Comment (terminaldweller) [reply .travis.yml:41]: not sure. is there any difference?

**Comment by terminaldweller** [comment]
im aware that the travis build is failing. fixing that right now.

**Comment by terminaldweller** [comment]
phew. its fixed now.

**Comment by terminaldweller** [comment]
added the badge also. it's good to go.

**Comment by chriseth** [comment]
Let me merge that so I can use the new testing script locally. I guess we should clean that up a little.

**Comment by terminaldweller** [comment]
yes, we need to clean stuff up.

**Pull request merged**